db/sql_implementation.go

```diff
@@ -69,25 +81,42 @@ func New(ctx context.Context, opts *Options) (*DB, error) {
 	connectCtx, cancel := context.WithTimeout(ctx, connectTimeout)
 	defer cancel()
 
-	if err := os.MkdirAll(filepath.Dir(opts.DataSourceName), os.ModePerm); err != nil && err != os.ErrExist {
-		return nil, err
+	pathsToCreate := []string{opts.DataSourceName, opts.DataSourceDHTName, opts.DataSourcePeerStoreName}
+	for _, path := range pathsToCreate {
+		if err := os.MkdirAll(filepath.Dir(path), os.ModePerm); err != nil && err != os.ErrExist {
+			return nil, err
+		}
 	}
 
 	sqldb, err := sqlx.ConnectContext(connectCtx, opts.DriverName, opts.DataSourceName)
```

Comment:
Is it possible to open these three using a for loop instead?

Reply:
We'd need either a `map[dbType]context` on the struct which would require changes to all uses of `db.sqldb` with a map getter, or add logic inside the loop to map it onto appropriate struct fields.

Since the number of contexts will most likely not change (`meshDB` , `peerstore`, `dht`) I think its fine if we keep this unrolled.